pkg/config/types/default_config.go

```diff
@@ -142,13 +145,36 @@ const defaultBacalhauDir = ".bacalhau"
 // 2. User's home directory with .bacalhau appended.
 // 3. If all above fail, use .bacalhau in the current directory.
 func DefaultDataDir() string {
+	// this method runs before root.go, so we set the level to info for these calls, then return it to previous value
+	currentLevel := zerolog.GlobalLevel()
+	zerolog.SetGlobalLevel(zerolog.InfoLevel)
+	defer zerolog.SetGlobalLevel(currentLevel)
+
+	// Check if the BACALHAU_DIR environment variable is set
 	if repoDir, set := os.LookupEnv("BACALHAU_DIR"); set && repoDir != "" {
 		return repoDir
+	} else if set {
+		log.Warn().Msg("BACALHAU_DIR environment variable is set but empty. Falling back to default directories.")
+	} else {
+		log.Debug().Msg("BACALHAU_DIR environment variable is not set. Trying to use $HOME.")
 	}
 
-	if userHome, err := os.UserHomeDir(); err == nil {
+	// Attempt to get the user's home directory
+	if userHome, err := os.UserHomeDir(); err == nil && filepath.IsAbs(userHome) {
+		log.Trace().Str("HomeDirectory", userHome).Msg("Successfully found $HOME. Using it for the data directory.")
 		return filepath.Join(userHome, defaultBacalhauDir)
+	} else {
+		log.Warn().Err(err).Msg("$HOME is unset or inaccessible. Falling back to current working directory.")
+	}
+
+	// Fallback: attempt to use the absolute path of the default directory
+	path, err := filepath.Abs(defaultBacalhauDir)
+	if err == nil {
+		log.Info().Str("Directory", path).Msg("Bacalhau will initialize in current working directory.")
+		return path
```

Comment:
I don't think we should. If no data directory was explicitly provided and user has no home, then we should just fail with a clear error message. 

Meaning this method should just return an empty data dir and let the config validation take care of the rest.

To be honest, and I think I've asked this multipe times before, I don't know why we are still looking into `BACALHAU_DIR` to get the default repo path. Default is `~/.bacalhau`, and then viper should take of the rest and replacing it with `BACALHAU_DIR` if it was provided